dimod/higherorder/utils.py

```diff
@@ -110,8 +112,8 @@ def reduce_binary_polynomial(poly: BinaryPolynomial) -> Tuple[
         ([(term, bias)*], [((orig_var1, orig_var2), aux_var)*])
 
     Example:
-        >>> poly = BinaryPolynomial({(0,): -1, (1,): 1, (2,): 1.5, (0, 1): -1, (0, 1, 2): -2}, dimod.BINARY)
-        >>> reduce_binary_polynomial(poly)
+        >>> poly = BinaryPolynomial({(0,): -1, (1,): 1, (2,): 1.5, (0, 1): -1, (0, 1, 2): -2}, dimod.BINARY) # doctest: +SKIP
```

Comment:
I would prefer to skip as little as possible. I think just `poly = dimod.BinaryPolynomail(...` would also avoid failures.

Reply:
Will trywith qualification. BinaryPolynomial is imported explicitly though.

Reply:
So the qualification does fix the first line of the doctest, but running
```
(cd docs; sphinx-build -b doctest . build/doctest)
```
locally gives me:
```
Document: reference/generated/dimod.higherorder.utils.reduce_binary_polynomial
------------------------------------------------------------------------------
**********************************************************************
File "../dimod/higherorder/utils.py", line ?, in default
Failed example:
    reduce_binary_polynomial(poly)
Exception raised:
    Traceback (most recent call last):
      File "/home/daniel/anaconda3/lib/python3.8/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest default[1]>", line 1, in <module>
        reduce_binary_polynomial(poly)
    NameError: name 'reduce_binary_polynomial' is not defined
**********************************************************************
1 items had failures:
   1 of   2 in default
2 tests in 1 items.
1 passed and 1 failed.
***Test Failed*** 1 failures.
```

Reply:
Ok, I was following up with @JoelPasvolsky , let's just leave it as-is for now. Can circle back around later.

Reply:
Ok I will only skip the second line then